sql/core/src/main/scala/org/apache/spark/sql/execution/command/AnalyzeColumnCommand.scala

```diff
@@ -0,0 +1,159 @@
+import org.apache.spark.sql.catalyst.plans.logical.{Aggregate, BasicColStats, Statistics}
+import org.apache.spark.sql.execution.datasources.LogicalRelation
+import org.apache.spark.sql.types._
+
+
+/**
+ * Analyzes the given columns of the given table in the current database to generate statistics,
+ * which will be used in query optimizations.
+ */
+case class AnalyzeColumnCommand(
+    tableIdent: TableIdentifier,
+    columnNames: Seq[String]) extends RunnableCommand {
+
+  override def run(sparkSession: SparkSession): Seq[Row] = {
+    val sessionState = sparkSession.sessionState
+    val relation = EliminateSubqueryAliases(sessionState.catalog.lookupRelation(tableIdent))
+
+    // check correctness of column names
+    val validColumns = mutable.MutableList[NamedExpression]()
+    val resolver = sessionState.conf.resolver
+    columnNames.foreach { col =>
+      val exprOption = relation.resolve(col.split("\\."), resolver)
+      if (exprOption.isEmpty) {
+        throw new AnalysisException(s"Invalid column name: $col")
+      }
+      if (validColumns.map(_.exprId).contains(exprOption.get.exprId)) {
+        throw new AnalysisException(s"Duplicate column name: $col")
+      }
+      validColumns += exprOption.get
+    }
+
+    relation match {
+      case catalogRel: CatalogRelation =>
+        updateStats(catalogRel.catalogTable,
+          AnalyzeTableCommand.calculateTotalSize(sparkSession, catalogRel.catalogTable))
+
+      case logicalRel: LogicalRelation if logicalRel.catalogTable.isDefined =>
+        updateStats(logicalRel.catalogTable.get, logicalRel.relation.sizeInBytes)
+
+      case otherRelation =>
+        throw new AnalysisException("ANALYZE TABLE is not supported for " +
+          s"${otherRelation.nodeName}.")
+    }
+
+    def updateStats(catalogTable: CatalogTable, newTotalSize: Long): Unit = {
+      // Collect statistics per column.
+      // The first element in the result will be the overall row count, the following elements
+      // will be structs containing all column stats.
+      // The layout of each struct follows the layout of the BasicColStats.
+      val ndvMaxErr = sessionState.conf.ndvMaxError
+      val expressions = Count(Literal(1)).toAggregateExpression() +:
+        validColumns.map(ColumnStatsStruct(_, ndvMaxErr))
+      val namedExpressions = expressions.map(e => Alias(e, e.toString)())
+      val statsRow = Dataset.ofRows(sparkSession, Aggregate(Nil, namedExpressions, relation))
+        .queryExecution.toRdd.collect().head
+
+      // unwrap the result
+      val rowCount = statsRow.getLong(0)
+      val colStats = validColumns.zipWithIndex.map { case (expr, i) =>
+        val colInfo = statsRow.getStruct(i + 1, ColumnStatsStruct.statsNumber)
+        val colStats = ColumnStatsStruct.unwrapRow(expr, colInfo)
+        (expr.name, colStats)
+      }.toMap
+
+      val statistics =
+        Statistics(sizeInBytes = newTotalSize, rowCount = Some(rowCount), basicColStats = colStats)
```

Comment:
One question. Seems we overwrite all existing column statistics, no matter whether we collect statistics of other columns before. Is it better to only replace the column statistics of the same columns? We can keep the column statistics of other columns. @wzhfy What do you think?